packages/vega-parser/src/parsers/guides/axis-config.js

```diff
@@ -1,12 +1,35 @@
 import {Bottom, Top} from './constants';
 import {extend} from 'vega-util';
+import { isSignal } from '../../util';
+import { xAxisExpr, axisOrientExpr } from './axis-util';
 
 export default function(spec, scope) {
   var config = scope.config,
       orient = spec.orient,
-      xy = (orient === Top || orient === Bottom) ? config.axisX : config.axisY,
-      or = config['axis' + orient[0].toUpperCase() + orient.slice(1)],
-      band = scope.scaleType(spec.scale) === 'band' && config.axisBand;
+      xy,
+      or,
+      band;
+
+  if (!isSignal(spec.orient)) {
+    xy = (orient === Top || orient === Bottom) ? config.axisX : config.axisY;
+    or = config['axis' + orient[0].toUpperCase() + orient.slice(1)];
+  } else {
+    var axisXYConfigKeys = new Set(Object.keys(config.axisX || {}).concat(Object.keys(config.axisY || {}))),
+    axisOrientConfigKeys = new Set(Object.keys(config.axisTop || {}).concat(Object.keys(config.axisBottom || {}).concat(Object.keys(config.axisLeft || {}).concat(Object.keys(config.axisRight ||{})))));
+
+
+    xy = {};
+    for (var prop of axisXYConfigKeys) {
+      xy[prop] = xAxisExpr(spec.orient.signal, config.axisX ? config.axisX[prop] : undefined, config.axisY ? config.axisY[prop] : undefined);
+    }
+
+    or = {};
+    for (prop of axisOrientConfigKeys) {
+      or[prop] = axisOrientExpr(spec.orient.signal, config.axisTop ? config.axisTop[prop] : undefined, config.axisBottom ? config.axisBottom[prop] : undefined, config.axisLeft ? config.axisLeft[prop] : undefined, config.axisRight ? config.axisRight[prop] : undefined);
+    }
+  }
+
+  band = scope.scaleType(spec.scale) === 'band' && config.axisBand;
```

Comment:
It doesn't seems like band depends on any of the logic above. 
If so, this could stay up where band is declared? 

(no need to cause unnecessary diff.)

Reply:
👍